src/runtime.ts

```diff
@@ -9,39 +9,47 @@ export { make, server, client, valueClass, reflection };
 export const noContentContentType = 'oatsNoContent' as const;
 export function noContent<Status extends number>(
   status: Status
-): server.Response<Status, typeof noContentContentType, null> {
+): server.Response<Status, typeof noContentContentType, null, null> {
   return {
     status,
     value: {
       contentType: noContentContentType,
       value: null
-    }
+    },
+    headers: null
   };
 }
 
-export function json<Status extends number, Value>(
+export function json<Status extends number, Value>(status: Status, value: Value) {
+  return jsonWithHeaders(status, value, null);
+}
+
+export function jsonWithHeaders<Status extends number, Value, Headers>(
```

Comment:
unsure on how to do this... feels ugly to add `jsonWithHeaders`, but also don't know if we want to just add a `headers` parameter to `json` and force people to always pass something (even if it's `null`). Looks like typescript doesn't like generics with default values very much